packages/origin-ui/src/__tests__/DemandForm.test.tsx

```diff
@@ -0,0 +1,249 @@
+                    .value
+            ).toBe('');
+
+            rendered.find(`#select-${name}`).simulate('click');
+
+            expect(
+                Array.from(document.querySelectorAll(`#menu-${name} ul li`)).map(i => i.textContent)
+            ).toStrictEqual(labels);
+
+            (document.querySelector(
+                `#menu-${name} [data-value="${valueToSelect}"]`
+            ) as HTMLElement).click();
+
+            await refresh();
+
+            expect(document.querySelector(`#menu-${name} [data-value="${valueToSelect}"]`)).toBe(
+                null
+            );
+
+            expect(
+                (rendered.find(`${dataTestSelector(name)} input`).getDOMNode() as HTMLInputElement)
+                    .value
+            ).toBe(valueToSelect);
+        }
+
+        await refresh();
+
+        const submitButton = rendered.find(`button${dataTestSelector('submitButton')}`);
+
+        expect(submitButton.text()).toBe('Create demand');
+        expect(submitButton.getDOMNode().hasAttribute('disabled')).toBe(true);
+
+        expect(
+            rendered
+                .find(`span${dataTestSelector('submitButtonTooltip')}`)
+                .getDOMNode()
+                .getAttribute('title')
+        ).toBe('Form needs to be valid to proceed.');
+
+        expect(rendered.find(dataTestSelector('totalDemand')).text()).toBe('0 MWh');
+
+        fillInputField('demandNeedsInMWh', '1');
+        fillInputField('maxPricePerMWh', '1');
+
+        await fillSelect('currency', 'EUR', ['EUR', 'USD', 'SGD', 'THB']);
+
+        await fillSelect('timeframe', TimeFrame.daily.toString(), ['Day', 'Week', 'Month', 'Year']);
+
+        await fillDate('startDate', 1);
+        await fillDate('activeUntilDate', 10);
+        await fillDate('endDate', 10);
+
+        expect(
+            rendered
+                .find(`span${dataTestSelector('submitButtonTooltip')}`)
+                .getDOMNode()
+                .getAttribute('title')
+        ).toBe(null);
+
+        expect(submitButton.getDOMNode().hasAttribute('disabled')).toBe(false);
+
+        createDemand = (offChainProps: Demand.IDemandOffChainProperties) => {
+            expect(moment.unix(parseInt(offChainProps.startTime, 10)).format('YYYY-MM-D')).toBe(
+                now
+                    .clone()
+                    .set('date', 1)
+                    .format('YYYY-MM-D')
+            );
+            expect(moment.unix(parseInt(offChainProps.endTime, 10)).format('YYYY-MM-D')).toBe(
+                now
+                    .clone()
+                    .set('date', 10)
+                    .format('YYYY-MM-D')
+            );
+
+            delete offChainProps.startTime;
+            delete offChainProps.endTime;
+
+            expect(offChainProps).toStrictEqual({
+                currency: Currency.EUR,
+                timeFrame: TimeFrame.daily,
+                maxPricePerMwh: 100,
+                targetWhPerPeriod: 1000000
+            });
```

Comment:
Maybe use Partial<Demand.IDemandOffChainProperties> instead of deleting startTime and endTime.

Reply:
Marking it as Partial wouldn't delete these properties - which is needed because strict comparing is performed between the arrays.

Reply:
Check this out https://github.com/energywebfoundation/origin/blob/50b4b8074e473c09636392b5b492df63c4b30ea3/packages/market/src/test/MarketFacade.test.ts#L265

Reply:
Aah, didn't know such function exists. Thanks @kosecki123